### PR TITLE
Add a test to verify batched dispatches result in a single render

### PR DESCRIPTION
**Pull request by lecstor**
Test file may need renaming/relocating.

Test does two separate rtl.act wrapped dispatches to verify that they result in 2 renders then repeats this wrapped in `batch`.

### CODE REVIEW

**Comment by lecstor** [comment]
@markerikson I thought I'd have a crack at this. `batch` looks good to me. Let me know what changes I need to make, or if I've missed the mark completely.

File: test/utils/batch.spec.js [hunk 1]
```diff
@@ -0,0 +1,94 @@
...
+import * as rtl from 'react-testing-library'
+import 'jest-dom/extend-expect'
+
+describe('React', () => {
+  describe('batch', () => {
+    afterEach(() => rtl.cleanup())
+
+    it('should render only once for batched dispatches', () => {
+      const store = createStore((state, action) => {
+        let newState =
```
Comment (markerikson) [inline test/utils/batch.spec.js]: Please extract the reducer as a separate variable, and use ES6 default arguments to initialize the state, like:

```js
const initialState = {a : 1};
const reducer = (state = initialState, action) => {}
```

**Comment by markerikson** [comment]
Seems like a good first check at behavior.  

Left one edit request for the code structure.

Also, could you add a second test for the `alternate-renderers` entry point? something like:

```js
import {batch as defaultBatch} from "../../../something/alternate-renderers";

// later
defaultBatch(() => {
    dispatch(increment())
    dispatch(increment())
})

expect(numberOfRenders).toBe(2)
```

**Comment by lecstor** [comment]
I see what you did there.. that `something` is more elusive than expected.

```
import { Provider as ProviderMock, batch, connect } from '../../src/index.js'
import { batch as defaultNoopBatch } from '../../src/alternate-renderers.js'
```
`Cannot find module '../../src/alternate-renderers.js' from 'batch.spec.js'`

**Comment by lecstor** [comment]
tests indicate that `defaultNoopBatch` is just as effective as the real thing.. I'm guessing the tests are flawed.

**Comment by lecstor** [comment]
copying the `defaultNoopBatch` function into the test and removing the import showed it worked as expected so with the combined tests/imports it seems the `defaultNoopBatch` was actually the real thing. Splitting the tests into separate files has resolved this.

**Comment by lecstor** [comment]
Should we be testing the render props are as expected, or is that out of scope?
eg
```
      expect(componentSpy).toHaveBeenLastCalledWith(
        {
          bar: 2,
          dispatch: store.dispatch,
          foo: 2
        },
        {}
      )
```

**Comment by timdorr** [comment]
Is this dead?

**Comment by lecstor** [comment]
I'm happy to make changes if needed, I was just waiting for feedback.

**Comment by markerikson** [comment]
I'll try to look at this over the weekend

**Comment by timdorr** [comment]
Hey @lecstor, I'm going to close this out since it's gotten a bit messy. Feel free to reopen it with a fresh branch and I'll take a look ASAP.

**Pull request closed without merging**